server/aws-lsp-codewhisperer/src/language-server/inline-completion/mergeRightUtils.ts

```diff
@@ -13,19 +18,115 @@ export function getPrefixSuffixOverlap(firstString: string, secondString: string
     return secondString.slice(0, i)
 }
 
-export function truncateOverlapWithRightContext(rightFileContent: string, suggestion: string): string {
+/**
+ * Returns the last index of the longest overlap between the first string and the second string
+ * @param targetString the target string
+ * @param searchString the search string
+ * @returns last index of the longest overlap between the first string and the second string
+ * @example getPrefixOverlapLastIndex("public static void", "static") = 13
+ */
+export function getPrefixOverlapLastIndex(targetString: string, searchString: string) {
+    let i = searchString.length
+    let idx = -1
+    while (i > 0) {
+        idx = targetString.indexOf(searchString.slice(0, i))
+        if (idx != -1) {
+            return idx + i
+        }
+        i--
+    }
+    return idx
+}
+
+export function truncateOverlapWithRightContext(
+    rightFileContent: string,
+    suggestion: string,
+    userEdit?: string
+): string {
```

Comment:
@atonaamz i just have a small request, as these "Edits" specific reconciliation logic is new, can we have a simple 
```
if (suggestionType === EDITS) {

} else {

}
```

to branch out these 2 types of suggestions and it can also ensure the "Completion" type suggestion wont regress?

Reply:
since these context merge logic is pretty delicate and will potentially require some iterations, it will be easier to branch out these 2 types of suggestions